great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2478,27 +2530,60 @@ def update_expectation_suite(
             )
 
         return self._add_expectation_suite(
-            expectation_suite_name=expectation_suite_name,
-            id=expectation_suite.ge_cloud_id,
-            expectations=expectation_suite.expectations,
-            evaluation_parameters=expectation_suite.evaluation_parameters,
-            data_asset_type=expectation_suite.data_asset_type,
-            execution_engine_type=expectation_suite.execution_engine_type,
-            meta=expectation_suite.meta,  # type: ignore[has-type] # Should just be a dict but mypy has trouble inferring
+            expectation_suite=expectation_suite,
             overwrite_existing=True,
         )
 
+    @overload
+    def add_or_update_expectation_suite(
+        self,
+        expectation_suite_name: str,
+        id: str | None = ...,
+        expectations: list[dict | ExpectationConfiguration] | None = ...,
+        evaluation_parameters: dict | None = ...,
+        data_asset_type: str | None = ...,
+        execution_engine_type: Type[ExecutionEngine] | None = ...,
+        meta: dict | None = ...,
+        expectation_suite: None = ...,
+    ) -> ExpectationSuite:
+        """
+        Two possible patterns:
+            - An expectation_suite_name and optional constructor args
+            - An expectation_suite
+        """
+        ...
+
+    @overload
+    def add_or_update_expectation_suite(
+        self,
+        expectation_suite_name: None = ...,
+        id: str | None = ...,
+        expectations: list[dict | ExpectationConfiguration] | None = ...,
+        evaluation_parameters: dict | None = ...,
+        data_asset_type: str | None = ...,
+        execution_engine_type: Type[ExecutionEngine] | None = ...,
+        meta: dict | None = ...,
+        expectation_suite: ExpectationSuite = ...,
+    ) -> ExpectationSuite:
+        """
+        Two possible patterns:
+            - An expectation_suite_name and optional constructor args
+            - An expectation_suite
```

Comment:
`non-blocking`
Just to make it quicker to grok. Rather than putting the same docstring on both overloads, could you describe which pattern each `overload` is meant to represent?

So this one would be something like..
```python
"""
An `expectation_suite` is provided.
`expectation_suite_name` should not be provided.
"""
```